source/connect/tls.txt

```diff
@@ -0,0 +1,259 @@
+or by using the ``tlsCAFile`` parameter in your connection string.
+
+.. include:: /includes/connect/ca-file-tabs.rst
+
+.. _cpp-specify-ca-directory:
+
+Specify a CA Directory
+~~~~~~~~~~~~~~~~~~~~~~
+
+If you are using OpenSSL or LibreSSL (``libtls``) for TLS support, you can also instruct
+the {+driver-short+} to search for a CA file within a directory. The driver searches this
+directory if it doesn't find a CA file at the path specified in the ``tlsCAFile`` or
+``pem_file`` option.
+ 
+The following code example shows how to use the ``ca_dir`` option to specify the
+directory that the driver should search:
+
+.. literalinclude:: /includes/connect/ca-dir.cpp
+      :language: cpp
+      :copyable: true
+
+.. tip:: 
+   
+   This option corresponds to the OpenSSL
+   `SSL_CTX_load_verify_locations <https://linux.die.net/man/3/ssl_ctx_load_verify_locations>`__
+   parameter and
+   the LibreSSL `tls_config_set_ca_path <https://man.openbsd.org/tls_load_file.3>`__
+   parameter. 
+
+.. _cpp-certificate-revocation:
+
+Check Certificate Revocation
+----------------------------
+
+When an X.509 certificate is no longer trustworthy—for example, if its private key
+has been compromised—the CA revokes the certificate. The {+driver-short+} includes two ways
+to check whether a server's certificate has been revoked.
+
+.. _cpp-disable-ocsp:
+
+OCSP
+~~~~
+
+The Online Certificate Status Protocol (OCSP) process varies depending on the version of
+{+mdb-server+} you're connecting to:
+
+- **MongoDB v4.4 or later:** The server staples a
+  time-stamped OCSP response to its certificate. The {+driver-short+} validates the certificate
+  against the OCSP response. If the CA has revoked the certificate, or if the OCSP response
+  is otherwise invalid, the TLS handshake fails.
+- **MongoDB v4.3 or earlier:** The server supplies an OCSP endpoint, which the {+driver-short+}
+  contacts directly. The {+driver-short+} then validates the certificate against the OCSP
+  response. If the CA hasn't revoked the certificate, the TLS handshake continues, even if
+  the OCSP response is invalid or malformed.
+
+To stop the {+driver-short+} from contacting the OCSP endpoint,
+set the ``tlsDisableOCSPEndpointCheck`` connection option to ``true`` in your
+connection string, as shown in the following code example:
+
+.. code-block:: cpp
+
+   mongocxx::uri uri("mongodb://<hostname>:<port>/?tls=true&tlsDisableOCSPEndpointCheck=true");
+
+.. _cpp-crl:
+
+Certificate Revocation List
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Instead of using OCSP, you can instruct the {+driver-short+} to check the server's certificate
+against a Certificate Revocation List (CRL) published by the CA.
+
+To do so, specify the path to a ``.pem`` or ``.der`` file from the CA.
+You can do this in two ways: by setting a property on a ``mongocxx::options::tls`` object,
+or by using the ``tlsCRLFile`` parameter in your connection string.
```

Comment:
> I didn't realize drivers were free to add new URI options beyond the spec.

I think the preference is to avoid driver-specific URI options so connection strings can be portable across drivers. `tlsCRLFile` option may have be an exception. [NODE-5549](https://jira.mongodb.org/browse/NODE-5549) notes: 

> Why isn't there a drivers spec option for this? Should there be one?
> Some driver SSL implementations do not support supplying a CRL